src/routes/api.svelte

```diff
@@ -0,0 +1,149 @@
+<script context="module">
+	import { backendUrl } from '$lib/config';
+</script>
+
+<script>
+	import Spinner from 'svelte-spinner';
+
+    export let submitting, submitted = false;
+	$: submitted = false;
+    export let apiRegistrationError = null;
+
+	async function handleSubmit(event) {
+        submitting = false;
+		const url = `${backendUrl}/register`;
+		let email = event.target.email.value;
+		let firstName = event.target.firstName.value;
+		let lastName = event.target.lastName.value;
+
+		// Avoid whitespace issues
+		email = email.trim();
+		firstName = firstName.trim();
+		lastName = lastName.trim();
+
+		submitting = true;
+
+        const params =  {
+            email,
+            first_name: firstName,
+            last_name: lastName
+        }
+
+        const encoded = new URLSearchParams(params)
+
+		try {
+			const res = await fetch(url, {
+				method: 'POST',
+				body: encoded
+			});
+
+			const data = await res.json();
+			if (!data.valid) {
+                apiRegistrationError = data.error;
+				return;
+			}
+
+            submitted = true;
```

Comment:
@kenkunz any clue about how i can indicated in green that the form was submitted, apparently the component is rerendering but I don't know how to catch the submitted state. Any help will be really appreciated.

Reply:
@servatj I pushed a few commits – sorry if I got carried away 🥺

2ad6b94a60c43 is the one that fixes the `submitted` status (renamed `success` in 0149b5188bf4)

See commit messages for details on other changes. On the whole, I tried to follow "Svelte best practice" approaches (e.g., using `bind:value` and local vars for state of inputs).  I also added some basic html5 form validations. I'm happy to talk though the changes if you'd like.

Reply:
Thanks a lot it looks great I think @miklko will be ready to merge this next week, I will try to learn more patterns and review all your commits so I 'm  more aligned with you. In my next ticket I will prepare the plan first and let you know before I start coding. I assume I won't work to much today an tomorrow. So we can speak next Monday, have a good weekend.